src/pages/App.tsx

```diff
@@ -209,7 +217,16 @@ export default function App() {
             <Suspense fallback={<Loader />}>
               {isLoaded ? (
                 <Routes>
-                  <Route path="/" element={<Landing />} />
+                  <Route
+                    path="/"
+                    element={
+                      // if react-router-dom matches "/" with window.location.hash defined, it means that we're
+                      // using BrowserRouter and can safely redirect to a path route
+                      <RedirectHashToPath>
```

Comment:
Why is this redirect necessary?

Reply:
If there's an old URL somewhere out there e.g. on Twitter with a hashtag, e.g. `uniswap.org/#/swap`, this would match the home page when BrowserRouter is used. Redirect will make sure that it is redirected to `uniswap.org/swap` before rendering home page.